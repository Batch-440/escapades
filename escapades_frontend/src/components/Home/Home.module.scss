.HomeImg {
  object-fit: cover;
}

.MainTitle {
  color: white;
  position: absolute;
  top: 33vh;
  left: 50%;
  transform: translate(-50%, 0);
  z-index: 9998;
  font-size: 3vw;
}
