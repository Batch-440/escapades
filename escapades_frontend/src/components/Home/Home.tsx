import classes from "./Home.module.scss";
import home from "../../assets/home.jpg"

export default function Home() {
  return (
    <div>
      <img className={classes.HomeImg} src={home}></img>
      <h1 className={classes.MainTitle}>Your next trip is waiting for you</h1>
    </div>
  )
}
