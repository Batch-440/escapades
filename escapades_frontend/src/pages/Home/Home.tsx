import CoverCard from "@/components/HomeBanner/HomeBanner";
import home from "@/assets/home.jpg";

const Home = () => {
  return (
    <>
      <CoverCard
        imageUrl={`url(${home})`}
        text={"Your next trip is waiting for you"}
      />
    </>
  );
};

export default Home;
