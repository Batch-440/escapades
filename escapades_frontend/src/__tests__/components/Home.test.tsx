import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Home from "@/pages/Home/Home";

describe("<Home />", () => {
  it("matches last screenshot", () => {
    const tree = renderer.create(<Home />);
    expect(tree).toMatchSnapshot();
  });

  it("should have title message visible", () => {
    render(<Home />);
    const title = screen.queryByText(/Your next trip is waiting for you/);
    expect(title).toBeVisible();
  });
});
