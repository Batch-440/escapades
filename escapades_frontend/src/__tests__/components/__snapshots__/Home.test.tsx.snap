// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Home /> > matches last screenshot 1`] = `
<div
  className="_imgContainer_0e51f0"
  style={
    {
      "backgroundImage": "url(/src/assets/home.jpg)",
    }
  }
>
  <h1
    className="_imgContainer__title_0e51f0"
  >
    Your next trip is waiting for you
  </h1>
</div>
`;
